src/Controller/ExportController.php

```diff
@@ -95,9 +103,43 @@ public function export(Request $request, ?SalesChannelContext $context): Respons
             return $errorResponse;
         }
 
+        $excludeProductGroups = filter_var($request->get('excludeProductGroups'), FILTER_VALIDATE_BOOLEAN);
+        $dynamicProductGroupService = $this->getDynamicProductGroupService();
+        if (!$excludeProductGroups && !$dynamicProductGroupService->isDynamicProductGroupWarmedUp()) {
+            $errorMessage = 'Dynamic Product Groups have not been warmed up yet. ';
+            $errorMessage .= 'This may cause missing categories! ';
+            $errorMessage .= "Warm them up by calling the route '/findologic/dynamic-product-groups', ";
+            $errorMessage .= 'or disable fetching of Dynamic Product Groups by adding the query parameter ';
+            $errorMessage .= "'excludeProductGroups=true'";
+
+            return new JsonResponse(['error' => $errorMessage], Response::HTTP_PRECONDITION_REQUIRED);
+        }
+
         return $this->doExport();
     }
 
+    /**
+     * @Route("/findologic/dynamic-product-groups", name="frontend.findologic.export.productgroup",
+     *     options={"seo"="false"}, methods={"GET"})
+     */
+    public function exportProductGroup(Request $request, ?SalesChannelContext $context): Response
+    {
+        $this->initialize($request, $context);
+        if ($errorResponse = $this->validate()) {
```

Comment:
When adding the validation for the dynamic product groups, you may want to add another method, that does simply that but without validating the set dynamic product groups. E.g. `$this->validateWithoutDynamicProductGroups()`. Feel free to change this name, as this name isn't very creative.